Display loading spinner instead of the (W) logo before site setup flow

p-jackson opened the pull request:
#### Changes proposed in this Pull Request

Still a work in progress. Ideally we will find a way to not have the CSS load on every page.
Change colour to grey30 if possible p1634273235332300/1634086694.194100-slack-C029SB8JT8S

* Adds `?loading-spinner` to the signup destination url when navigating to the `site-setup` flow
* SSR will show a spinner instead of the (W) logo when the `loading-spinner` query param is present

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run `yarn start` again to rebuild the node server

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

p-jackson committed:
Display loading spinner instead of the (W) logo before site setup flow

roo2 commented on client/signup/config/flows.js:
```diff
@@ -61,7 +61,7 @@ function getSignupDestination( { domainItem, siteId, siteSlug } ) {
 		return '/home';
 	}
-	let queryParam = { siteSlug };
+	let queryParam = { siteSlug, 'loading-spinner': 1 };
```
I think we'd usually use snake_case or camel case for query params

roo2 commented on client/document/index.jsx:
```diff
@@ -245,4 +246,30 @@ class Document extends Component {
+	return WordPressLogo;
+}
+
+function Spinner( { className } ) {
```
We should probably implement this as a calypso component that can be shared with the reskinned processing screen.

arthur791004 replied:
It's great now but it might be better to make it a general component and we can re-use this one between here and reskinned flow.

roo2 reviewed (commented):
I think now we can break out a component that can be shared with the processing screen work https://github.com/Automattic/wp-calypso/pull/57051/files

p-jackson committed:
Center the loading ellipsis on the processing screen

This will make it easier to align with the ellipsis rendered by the
server during loading.

p-jackson committed:
Align loading animation in the center

p-jackson committed:
Rename query param from loading-ellipsis to loading_ellipsis

arthur791004 reviewed (approved):
LGTM 👍
